src/utils/rules.ts

```diff
@@ -66,6 +71,42 @@ export const styleColor = ({ prop, cssProperty }: { prop: string; cssProperty?:
   return fn;
 };
 
+export const style = ({
+  prop,
+  cssProperty,
+  key,
+  transformValue,
+}: {
+  prop: string;
+  cssProperty?: string;
+  key?: string;
+  transformValue?: Function;
+  // TODO FIXME
+}): TemplateStringsArray => {
+  const cssName = cssProperty || prop;
+  const fn = (props: any) => {
+    let val = props[prop];
+    if (!is(val)) return null;
```

Comment:
We have a similar functionality in `styleColor`. Could we reuse it or extract to some shared util function?

Reply:
Similar but different, in `styleColor` there is some recursive action to resolve json refs in the theme so that colors can reference eachother. Although that functionality might make sense to have always.  Will see if there is a smart way to handle!

Reply:
yeah they're too different to combine them

Reply:
Found a solution to combine!